Windows release build not building

I just tried to build in Release mode on Windows with MSVC 2019 and it failed with:

```
mavlink_mission_transfer_test.obj : error LNK2001: unresolved external symbol "class testing::internal::Mutex testing::
internal::g_gmock_mutex" (?g_gmock_mutex@internal@testing@@3VMutex@12@A) [C:\Users\Julian Oes\src\MAVSDK\build\release\
src\unit_tests_runner.vcxproj]
mavlink_mission_transfer_test.obj : error LNK2001: unresolved external symbol "class testing::internal::ThreadLocal<cla
ss testing::Sequence *> testing::internal::g_gmock_implicit_sequence" (?g_gmock_implicit_sequence@internal@testing@@3V?
$ThreadLocal@PEAVSequence@testing@@@12@A) [C:\Users\Julian Oes\src\MAVSDK\build\release\src\unit_tests_runner.vcxproj]
C:\Users\Julian Oes\src\MAVSDK\build\release\src\Debug\unit_tests_runner.exe : fatal error LNK1120: 2 unresolved extern
als [C:\Users\Julian Oes\src\MAVSDK\build\release\src\unit_tests_runner.vcxproj]
```

--- a/src/cmake/unit_tests.cmake
+++ b/src/cmake/unit_tests.cmake
@@ -26,4 +26,11 @@ target_link_libraries(unit_tests_runner
     gmock
 )
 
+if (MSVC AND BUILD_SHARED_LIBS)
+    target_compile_definitions(unit_tests_runner PRIVATE -DGTEST_LINKED_AS_SHARED_LIBRARY)
+    set_target_properties(unit_tests_runner
+        PROPERTIES COMPILE_FLAGS "${warnings} /wd4275"
+    )
+endif()
+
 add_test(unit_tests unit_tests_runner)

